pkg/util/slice_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "testing"
+
+func TestCompareStringSlice(t *testing.T) {
+	tests := []struct {
+		a  []string
+		b  []string
+		ok bool
+	}{
+		{[]string{}, []string{}, true},
+		{[]string{"", ""}, []string{""}, false},
+		{[]string{"", ""}, []string{"", ""}, true},
+		{[]string{"a", "b"}, []string{"a", "b"}, true},
+		{[]string{"a", "b"}, []string{"b", "a"}, false},
+		{[]string{"a", "b", "c"}, []string{"a", "b", "c"}, true},
+		{[]string{"a", "b", "c"}, []string{"b", "c", "a"}, false},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
+	}
+
+	for i, tt := range tests {
+		ok := CompareStringSlice(tt.a, tt.b)
+		if ok != tt.ok {
+			t.Errorf("%d: got %t but wanted: %t a: %v, b: %v", i, ok, tt.ok, tt.a, tt.b)
+		}
+	}
+}
+
+func TestCompareStringSliceNoOrder(t *testing.T) {
+	tests := []struct {
+		a  []string
+		b  []string
+		ok bool
+	}{
+		{[]string{}, []string{}, true},
+		{[]string{"", ""}, []string{""}, false},
+		{[]string{"", ""}, []string{"", ""}, true},
+		{[]string{"a", "b"}, []string{"a", "b"}, true},
+		{[]string{"a", "b"}, []string{"b", "a"}, true},
+		{[]string{"a", "b", "c"}, []string{"a", "b", "c"}, true},
+		{[]string{"a", "b", "c"}, []string{"b", "c", "a"}, true},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
```

Comment:
May want to add some cases where the two slices have different length to get coverage of the `if len(a) != len(b) { return false }` bit (which is what makes it correct e.g. for `{abc, abcd, false}`).

Reply:
It's covered by the second test (the one with empty strings) but I'll add a more explicit test with non empty strings to make it clear.